pkg/inject/secretless.go

```diff
@@ -1,55 +1,120 @@
-                Image:           "cyberark/secretless-broker:latest",
-                Args:            []string{"-f", "/etc/secretless/secretless.yml"},
-                ImagePullPolicy: "Always",
-                VolumeMounts: []corev1.VolumeMount{
-                    {
-                        Name:      "secretless-config",
-                        ReadOnly:  true,
-                        MountPath: "/etc/secretless",
-                    },
-                },
-                Env: envvars,
-            },
-        },
-        Volumes: []corev1.Volume{
-            {
-                Name: "secretless-config",
-                VolumeSource: corev1.VolumeSource{
-                    ConfigMap: &corev1.ConfigMapVolumeSource{
-                        LocalObjectReference: corev1.LocalObjectReference{
-                            Name: secretlessConfigMapName,
-                        },
-                    },
-                },
-            },
-        },
-    }
+func generateSecretlessSidecarConfig(secretlessConfig, conjurConnConfigMapName, conjurAuthConfigMapName, ServiceAccountTokenVolumeName string) *PatchConfig {
+	envvars := []corev1.EnvVar{
+		envVarFromFieldPath("MY_POD_NAME", "metadata.name"),
+		envVarFromFieldPath("MY_POD_NAMESPACE", "metadata.namespace"),
+		envVarFromFieldPath("MY_POD_IP", "status.podIP"),
+	}
+
+	if conjurConnConfigMapName != "" || conjurAuthConfigMapName != "" {
+		envvars = append(envvars,
+			envVarFromConfigMap("CONJUR_VERSION", conjurConnConfigMapName),
+			envVarFromConfigMap("CONJUR_APPLIANCE_URL", conjurConnConfigMapName),
+			envVarFromConfigMap("CONJUR_AUTHN_URL", conjurConnConfigMapName),
+			envVarFromConfigMap("CONJUR_ACCOUNT", conjurConnConfigMapName),
+			envVarFromConfigMap("CONJUR_SSL_CERTIFICATE", conjurConnConfigMapName),
+			envVarFromConfigMap("CONJUR_AUTHN_LOGIN", conjurAuthConfigMapName))
+	}
+
+	// Allow configmgr#configspec in the SecretlessConfig annotation
+	var configMgr string
+	var configSpec string
+	var secretlessConfigMapName string
+	secretlessConfigPath := "/etc/secretless/secretless.yml"
+	volumes := []corev1.Volume{}
+
+	// Always add Service Account Token Volume Mount (SATVM)
+	// It shouldn't be sidecar-injector's responsibility to add the SATVM, we only
+	// do it here because the serviceaccount plugin which adds the SATVM is
+	// executed before this plugin injects the sidecar. KEP-36 will solve this
+	// ordering problem by re-running plugins after mutation. Then if we add a
+	// container, the serviceaccount plugin will reprocess the manifest and add the
+	// appropriate mount
+	//
+	// ** Remove me when KEP-36 lands **
+	// also remove common.getServiceAccountTokenVolumeName
+	// and calls to that in server.go
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      ServiceAccountTokenVolumeName,
+			ReadOnly:  true,
+			MountPath: "/var/run/secrets/kubernetes.io/serviceaccount",
+		},
+	}
+
+	// Three options for secretlessConfig
```

Comment:
I like that this maintains backwards compatibility :)... but feel free to rip apart the contract and start from scratch if some other interface makes more sense. `Naming and functionality are still subject to breaking changes.`

Reply:
Good point, but left as-is for now, this patch has been in progress too long already :)